src/test/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplicationSuite.scala

```diff
@@ -0,0 +1,906 @@
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.armada.submit
+
+import api.submit.JobSubmitRequestItem
+import k8s.io.api.core.v1.generated.{EnvVar, PodSpec, Volume, VolumeMount}
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.armada.Config
+import org.apache.spark.deploy.armada.submit.{ConfigGenerator, JobTemplateLoader}
+import org.scalatest.BeforeAndAfter
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+
+import java.io.{File, FileWriter}
+import java.nio.file.{Files, Path}
+import scala.concurrent.duration._
+
+class ArmadaClientApplicationSuite extends AnyFunSuite with BeforeAndAfter with Matchers {
+
+  private var sparkConf: SparkConf = _
+  private val armadaClientApp      = new ArmadaClientApplication()
+  private var tempDir: Path        = _
+
+  before {
+    tempDir = Files.createTempDirectory("armada-client-test-")
+    sparkConf = new SparkConf()
+      .set("spark.master", "armada://localhost:50051")
+      .set("spark.app.name", "test-app")
+      .set(Config.ARMADA_JOB_QUEUE.key, "test-queue")
+      .set(Config.ARMADA_JOB_SET_ID.key, "test-job-set")
+      .set(Config.ARMADA_SPARK_JOB_NAMESPACE.key, "test-namespace")
+      .set(Config.ARMADA_SPARK_JOB_PRIORITY.key, "1.0")
+      .set(Config.CONTAINER_IMAGE.key, "spark:3.5.0")
+      .set(Config.ARMADA_JOB_NODE_SELECTORS.key, "node-type=compute")
+  }
+
+  after {
+    if (tempDir != null) {
+      tempDir.toFile.listFiles().foreach(_.delete())
+      Files.deleteIfExists(tempDir)
+    }
+  }
+
+  private def createJobTemplateFile(queue: String, jobSetId: String): File = {
+    val templateContent =
+      s"""queue: $queue
+         |jobSetId: $jobSetId
+         |jobRequestItems:
+         |  - priority: 1.5
+         |    namespace: template-namespace
+         |    labels:
+         |      template-label: template-value
+         |    annotations:
+         |      template-annotation: template-value
+         |""".stripMargin
+
+    val file   = tempDir.resolve("job-template.yaml").toFile
+    val writer = new FileWriter(file)
+    try {
+      writer.write(templateContent)
+    } finally {
+      writer.close()
+    }
+    file
+  }
+
+  private def createJobItemTemplateFile(
+      priority: Double,
+      namespace: String,
+      filename: String = "job-item-template.yaml"
+  ): File = {
+    val templateContent =
+      s"""priority: $priority
+         |namespace: $namespace
+         |labels:
+         |  item-template-label: item-template-value
+         |annotations:
+         |  item-template-annotation: item-template-value
+         |""".stripMargin
+
+    val file   = tempDir.resolve(filename).toFile
+    val writer = new FileWriter(file)
+    try {
+      writer.write(templateContent)
+    } finally {
+      writer.close()
+    }
+    file
+  }
+
+  test("validateArmadaJobConfig should create config without templates") {
```

Comment:
These unit tests look like they are going to be really good for the project, but they are a bit hard to review.

In order to review them properly, I'd like to propose the following:
1. All string config values that come from the runtime should have the word runtime in it.  All those that come from a template should have the word template in them.
2. If they are not string values, then I'd like to see them defined with names that include "RUNTIME" or "TEMPLATE"

What do you think?